### ISSUE TITLE
Don't use sprintf

### ISSUE THREAD

**Issue by bangerth** [GeodynamicWorldBuilder/WorldBuilder#731]
The ASPECT logs show these warnings:
```
/Users/jenkins/jenkins/workspace/aspect-osx_PR-5715/contrib/world_builder/include/doctest/doctest.h:4158:3: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  DOCTEST_TO_STRING_OVERLOAD(char, "%d")
  ^
/Users/jenkins/jenkins/workspace/aspect-osx_PR-5715/contrib/world_builder/include/doctest/doctest.h:4154:10: note: expanded from macro 'DOCTEST_TO_STRING_OVERLOAD'
    std::sprintf(buf, fmt, in);                                                                \
         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
```
I don't know whether doctest is under your control, but it's probably worth addressing these warnings.

**Comment by MFraters** [comment]
Thanks for reporting. Yes, this can be changed in the world builder, and then be upstreamed if needed. I will take a look at this.

**Comment by MFraters** [comment]
I noticed this has already been resolved in #704. I am making a backport to ASPECT.
